library/ssl_misc.h

```diff
@@ -1829,7 +1829,13 @@ void mbedtls_ssl_set_timer(mbedtls_ssl_context *ssl, uint32_t millisecs);
 MBEDTLS_CHECK_RETURN_CRITICAL
 int mbedtls_ssl_check_timer(mbedtls_ssl_context *ssl);
 
-void mbedtls_ssl_reset_in_out_pointers(mbedtls_ssl_context *ssl);
+void mbedtls_ssl_reset_in_pointers(mbedtls_ssl_context *ssl);
+void mbedtls_ssl_reset_out_pointers(mbedtls_ssl_context *ssl);
+static inline void mbedtls_ssl_reset_in_out_pointers(mbedtls_ssl_context *ssl)
```

Comment:
```suggestion
void mbedtls_ssl_reset_in_out_pointers(mbedtls_ssl_context *ssl);
```

It makes sense to keep the existing method, while enhancing the granularity when we need to reset only one type of pointer. 

This also means that line [389 ](https://github.com/Mbed-TLS/mbedtls/blob/5f7c2c21825518d87912aafc8ad5bda5ad0f320b/library/ssl_tls.c#L389-L390) in `library/ssl_tls.c` can remain unmodified.

Reply:
sorry, i'm afraid i don't quite understand your suggestion.

we need to separate resetting in and out pointers, so i split the function in two - such is the case of handle_buffer_resizing (line ssl_tls.c 389 you mentioned):  we still _reset_ the out pointers but we _update_ in pointers: since we may have fragments buffered we don't want in_hdr reset to the beginning of the buffer,  so it's not quite the same and we can't reset both at this point.

we still reset both in two cases, and i did keep mbedtls_ssl_reset_in_out_pointers, just converted it to static inline, since it's literally two lines: reset in, reset out. so... done?

maybe i'm misunderstanding the suggestion, can you please clarify?

Reply:
Sorry I also did not make myself very clear. 

It may be prefferably to remove `mbedtls_ssl_reset_in_out_pointers completely`, and invoke the two lines in the two remaining occurences in code just like you do [here](https://github.com/Mbed-TLS/mbedtls/blob/5f7c2c21825518d87912aafc8ad5bda5ad0f320b/library/ssl_tls.c#L389-L390)

If you want you could also add it as a static inline in the `ssl_tls.c` but honestly I do not think it is worth the effort. 

The library is being compiled using a variety of compilers, and linked with cpp as well as c code, and `static inline` in header files is unfortunately not handled in a standard way.

Reply:
ok, sorry, i misunderstood. sure, makes sense - done.